### PR TITLE
chore(plugins): moves UI of friends and activity to plugins

### PR DESCRIPTION
**Pull request by mrclay**
Site owners can disable group activity pages site-wide.

Adds hook `filter_tabs` to allow modifying default tabs.

Much of the views, actions, language files are left in core for simplicity. These can be addressed later.

These will be moved to separate repos with the others.
- [x] document hook `filter_tabs`.

### CODE REVIEW

**Comment by mrclay** [comment]
@hypeJunction this would certainly conflict with your work. I'm happy to let this sit until yours is in.

**Comment by hypeJunction** [comment]
Yes, let's wait. Plus I don't think ee should be leaving views and callbacks behind - either move everything or keep.

**Comment by mrclay** [comment]
We should establish the goals first. My only goal was to make it easy to disable these features. That really could be done with a couple settings.

The basic mechanics of friends and rivers I think could be something Elgg core stays opinionated about.

**Comment by hypeJunction** [comment]
> The basic mechanics of friends and rivers I think could be something Elgg core stays opinionated about.

I can get behind that.

**Comment by hypeJunction** [comment]
Needs rebase

**Comment by mrclay** [comment]
Rebased.

File: views/default/forms/admin/site/advanced/content_access.php [hunk 1]
```diff
@@ -11,6 +12,20 @@
 		ACCESS_LOGGED_IN => elgg_echo('LOGGED_IN'),
 		ACCESS_PUBLIC => elgg_echo('PUBLIC'),
 	],
+=======
```
Comment (mrclay) [inline views/default/forms/admin/site/advanced/content_access.php]: oops

**Comment by mrclay** [comment]
@hypeJunction I reworked `elgg_get_filter_tabs` so it's hook-alterable rather than use the hacky `elgg_is_active_plugin('friends')` check.

**Review by hypeJunction — APPROVED** [review]
Lgtm

**Comment by jdalsem** [comment]
isn't the commit message a feature?

This would fix #3445

**Comment by jdalsem** [comment]
There is also a lot of activity related code in the groups plugin

**Comment by mrclay** [comment]
@jdalsem I will add that

**Comment by mrclay** [comment]
I think we'd want to just add a global groups setting on whether to allow groups to add activity. Does that sound right?

**Comment by jdalsem** [comment]
> I think we'd want to just add a global groups setting on whether to allow groups to add activity. Does that sound right?

that is something we do not have now right? Groups can individually set the group_tool activity right now... so a plugin setting to enable/disable activity for all groups is what you are suggesting?

**Comment by mrclay** [comment]
> so a plugin setting to enable/disable activity for all groups is what you are suggesting?

Yes. Added to this PR.

File: engine/lib/river.php [hunk 2]
```diff
@@ -755,14 +726,9 @@ function _elgg_river_enable($event, $type, $entity) {
  * @access private
  */
 function _elgg_river_init() {
-	elgg_register_page_handler('activity', '_elgg_river_page_handler');
-	$item = new \ElggMenuItem('activity', elgg_echo('activity'), 'activity');
-	elgg_register_menu_item('site', $item);
 
 	elgg_register_widget_type('river_widget', elgg_echo('river:widget:title'), elgg_echo('river:widget:description'));
```
Comment (jdalsem) [inline engine/lib/river.php]: why not move the widget?? this is also UI related

File: mod/friends/start.php [hunk 3]
```diff
@@ -231,10 +225,68 @@ function _elgg_send_friend_notification($event, $type, $object) {
...
+ * @param string $hook         "access:collections:write"
+ * @param string $type         "user"
+ * @param array  $access_array Access array
+ * @param array  $params       Hook params
+ *
+ * @return array
+ */
+function _elgg_friends_write_access($hook, $type, $access_array, $params) {
+
+	// rebuild array, putting friends 1st or 2nd
```
Comment (jdalsem) [inline mod/friends/start.php]: is this really necessary? MySQL will optimize query to run the fastest am i right? Did you run tests to confirm this reorder is needed?

Comment (mrclay) [inline mod/friends/start.php]: Without this, the "Friends" option would go below all the other options.

**Review by hypeJunction — APPROVED** [review]

**Comment by hypeJunction** [comment]
@mrclay you might eant to pull the plug in this before it goes stale

**Pull request merged**